src/50expression.js

```diff
@@ -688,18 +688,33 @@
 			assign(this, params);
 		}
 
-		toString() {
-			let s = this.columnid;
+		// Check if identifier needs to be wrapped in brackets
+		// (contains spaces, dots, or other special characters)
+		_needsBrackets(id) {
+			if (id == null) return false;
+			// Numeric indices need brackets
+			if (id == +id) return true;
+			// Check for special characters that require brackets (spaces, dots, hyphens, square brackets)
+			return /[\s.\-\][]/.test(id);
+		}
 
-			if (this.columnid == +this.columnid) {
-				s = '[' + this.columnid + ']';
+		_wrapId(id) {
+			if (this._needsBrackets(id)) {
+				return '[' + id + ']';
```

Comment:
is there a reason we dont use ``` ` ``` but ``` [ ``` ?

Im not asking you to change it. Im asking why you picked {